packages/frontend/src/pages/Flows/index.tsx

```diff
@@ -84,10 +84,16 @@ export default function Flows(): React.ReactElement {
 
   const { pageInfo, edges } = data?.getFlows || {}
   const flows: IFlow[] = edges?.map(({ node }: { node: IFlow }) => node) ?? []
-  const hasPagination =
-    !loading && pageInfo && pageInfo.totalCount > FLOW_PER_PAGE
+  const totalCount: number = pageInfo?.totalCount ?? 0
+  const hasPagination = !loading && totalCount > FLOWS_PER_PAGE
   const hasNoUserFlows = flows.length === 0 && !isSearching
 
+  // ensure invalid pages won't be accessed even after deleting flows
+  const lastPage = Math.ceil(totalCount / FLOWS_PER_PAGE)
+  if (lastPage !== 0 && page > lastPage) {
+    setSearchParams({ page: lastPage })
+  }
+
```

Comment:
`lastPage` will only be 0 if totalCount is 0 right?

Reply:
unrelatedly, manually accessing ?page=999 will cause this error

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/ycT2Xb5nBEjN2kf2gy95/5866ab99-436d-41b9-bbdb-14d163f4e009.png)

this is due to setting useState during render.

a simple fix is to wrap it in useEffect

Reply:
oh okkk got it, i used it initially but thought it was unnecessary, now i see why i needed it thankss